### PR TITLE
Add support for required reviews

### PR DESCRIPTION
**Pull request by stephtr**
This PR adds a new parameter, `MERGE_REQUIRED_APPROVALS` with which the minimal amount of required PR review approvals can be specified.

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -118,9 +118,9 @@ async function handleArbitraryPullRequestUpdate(context, eventData) {
       pull_number: pullRequestNumber
     });
     logger.trace("Full PR:", pullRequest);
-
+    const approvalCount = await fetchApprovalReviewCount(context, pullRequest);
     await update(context, pullRequest);
-    await merge(context, pullRequest);
+    await merge(context, pullRequest, approvalCount);
```
Comment (pascalgn) [inline lib/api.js:123]: Maybe instead of passing it to all the merge calls, do the fetchApprovalReviewCount call in the merge() function itself?

**Comment by pascalgn** [comment]
GH has this feature (require a certain amount of approvals) built-in:
image

Could you give some motivation for adding it to this action? 🙏

File: dist/index.js [hunk 2]
```diff
@@ -414,6 +434,23 @@ async function fetchPullRequest(context, repository, issue) {
...
+  const { number } = pullRequest;
+
+  logger.debug("Getting reviews for", number, "...");
+  let { data: reviews } = await octokit.pulls.listReviews({
+    owner: pullRequest.base.repo.owner.login,
+    repo: pullRequest.base.repo.name,
+    pull_number: number
+  });
+
+  reviews = reviews.filter(review => review.state === "APPROVED");
```
Comment (pascalgn) [inline dist/index.js:448]: When we wanted to see if everyone in the team is roughly doing the same amount of code reviews, someone figured out that GH allows you to give multiple approvals for the same PR, which quickly lead to our statistic being "slightly" skewed until we fixed the measurement script. So here we should probably only count the distinct authors, not the total number of approvals.

**Comment by stephtr** [comment]
Thank you for your feedback!
My main motivation for this work (I already use it on my repos) is that the branch protection feature isn't included with GitHub's free organization accounts.
I'll try to address your comments within the next few days.

File: dist/index.js (hunk 2 above)
Comment (stephtr) [reply dist/index.js:448]: That's a good catch, I made the wrong assumption.

File: lib/api.js (hunk 1 above)
Comment (stephtr) [reply lib/api.js]: To me the merge function felt like the wrong spot to fetch new data (that would also complicate mocking).
In order to not repeat the code over and over, I combined the calls to `fetchApprovalReviewCount`, `update` and `merge` into a helper function instead.

**Comment by stephtr** [comment]
Sorry for the delay and thanks for the idea for the two changes! It's updated now.

**Comment by pascalgn** [comment]
Thanks for the PR, it's merged now! 🎉 👍

**Pull request merged**